### ISSUE TITLE
Fix "PHPUnit_Framework_TestCase::getMock() is deprecated" warnings

### ISSUE THREAD

**Issue by Potherca** [ramsey/uuid#186]
Since version **3.5** (commit 22226169cd8bd77d2b9e7882e2b037c1fa964197) the `composer.js` version constraint for `phpunit/phpunit` (in the `require-dev` section) has been limited to `<5.4`  in order to to avoid deprecation warnings.

Rather than limit the PHPUnit version, wouldn't it be better to simply fix the calls it complains about?

If this suggestion meets your approval, I have [code changes I can submit a PR for](https://github.com/potherca-contrib/uuid/compare/master...fix/replace-deprecated-getMock).

**Comment by ramsey** [comment]
Since the library still supports PHP 5.4, will these changes pass on PHPUnit versions that get installed on PHP 5.4-5.6?

**Comment by ramsey** [comment]
BTW, the next major version of ramsey/uuid will target PHP 7+ only, in case that matters for your changes.

**Comment by Potherca** [comment]
The changes will work across PHPUnit 4, 5 and 6 (which is needed if test are run under PHP7) as I've only changed calls to`Testcase::getMock()` with calls to `Testcase::getMockBuilder()->getMock()`.

In regards to PHP7 only, several things in PHPUnit have been deprecated. 
I've already created [an inventory of potential problems](https://gist.github.com/Potherca/1533bde47294a2bb860b18431d2c6291) for several projects I am involved in.

It is possible to create tests that run under both PHP5 and PHP7 (i.e. PHPUnit 5 and 6) but if only PHPUnit 6 needs to be supported it is simply a matter of replacing the functions that have been deprecated in PHPUnit 5 with the (forward compatible) PHPUnit 6 features.

I could open a separate PR for that too, if you'd like one. :+1:
